lib/LibeventReactor.php

```diff
@@ -60,7 +60,7 @@ public function run(callable $onStart = null) {
         }
 
         while ($this->isRunning) {
-            if ($this->immediates && !$this->doImmediates()) {
+            if (!$this->doImmediates()) {
```

Comment:
After re-reading the PR desc, I got it. You expect new immediates get generated while the loop